androidApp/src/main/java/com/mbta/tid/mbta_app/android/map/HomeMapView.kt

```diff
@@ -242,11 +244,11 @@ fun HomeMapView(
             LaunchedEffect(railRouteShapes, globalResponse, globalMapData) {
                 viewModel.refreshRouteLineData(now)
             }
-            LaunchedEffect(railRouteLineData) {
+            LaunchedOnDispatcherEffect(Dispatchers.Default, railRouteLineData) {
```

Comment:
I like the way you broke out reusable helpers, but I'm concerned about how we would consistently use these new helpers when appropriate. 

Wrapping my head more on coroutines, I found a couple of snippets from [this blog post](https://medium.com/androiddevelopers/coroutines-on-android-part-i-getting-the-background-3e0e54d20bb) helpful:

> a good practice is to use withContext to make sure every function is safe to be called on any Dispatcher including Main — that way the caller never has to think about what thread will be needed to execute the function.

> It’s a really good idea to make every suspend function main-safe. If it does anything that touches the disk, network, or even just uses too much CPU, use withContext to make it safe to call from the main thread. This is the pattern that coroutines based libraries like Retrofit and Room follow. If you follow this style throughout your codebase your code will be much simpler and avoid mixing threading concerns with application logic. When followed consistently, coroutines are free to launch on the main thread and make network or database requests with simple code while guaranteeing users won’t see “jank.”

What do you think of pushing the Dispatcher context to a lower level, such as within `refreshRouteLineSource` and `refreshStopFeatures` so that the view doesn't need to worry about thread management?

Aside from making any changes to the implementation here, I think it would be worth adding a note to the PR template to check that expensive computations are run on the Default thread as appropriate.